lib/screens/authentication/recover/recover_account_found/interactor/mappers/fetch_recover_guardian_state_mapper.dart

```diff
@@ -49,8 +48,12 @@ class FetchRecoverRecoveryStateMapper extends StateMapper {
         recoveryStatus = RecoveryStatus.WAITING_FOR_GUARDIANS_TO_SIGN;
       }
 
-      // Save the private key and account
-      SaveAccountUseCase().run(accountName: currentState.userAccount, privateKey: result.privateKey);
+      // (Gery)
```

Comment:
if its already saved, no.

Reply:
Yes, we need to save the account and private key in this step

Reply:
if (settingsStorage.privateKey != null && settingsStorage.inRecoveryMode) {
      recoveryPrivateKey = settingsStorage.privateKey!;
    } else {
      recoveryPrivateKey = EOSPrivateKey.fromRandom().toString();
      settingsStorage.enableRecoveryMode(accountName: accountName, privateKey: recoveryPrivateKey);
    }
    
    This code in the `FetchRecoverGuardianInitialDataUseCase` takes care of saving the key if its first time getting in recovery mode. 
    
    And if its not first time, then the key is already saved. 
    
    So no. This line of code saving the key here is not needed.

Reply:
Nice I see also the accountName is saved the first time.

Reply:
Removed 👍